scipy/interpolate/tests/test_rgi.py

```diff
@@ -621,93 +691,142 @@ def test_xi_nd(self):
+        # NB: values has a `num_trailing_dims` trailing dimension
+        values = rng.random((6, 7, 8, 9, *trailing_points))
+        sample = rng.random(4)   # a single sample point !
 
-            assert_allclose(v, v2, err_msg=method)
+        v = interpn(points, values, sample, method=method, bounds_error=False)
 
+        # v has a single sample point *per entry in the trailing dimensions*
+        assert v.shape == (1, *trailing_points)
+
+        # check the values, too : manually loop over the trailing dimensions
+        vs = [interpn(points, values[..., j], sample, method=method,
+                      bounds_error=False) for j in range(values.shape[-1])]
+
+        # need to transpose otherwise first dimension is in the wrong place
+        axes = tuple(range(1 + num_trailing_dims))
+        axx = axes[1:] + axes[:1]
+        v2 = np.asarray(vs).transpose(axx)
+
+        assert_allclose(v, v2, atol=1e-14, err_msg=method)
+
+    def test_non_scalar_values_splinef2d(self):
         # Vector-valued splines supported with fitpack
+        points, values = self._sample_4d_data()
+
+        np.random.seed(1234)
+        values = np.random.rand(3, 3, 3, 3, 6)
+        sample = np.random.rand(7, 11, 4)
         assert_raises(ValueError, interpn, points, values, sample,
                       method='splinef2d')
 
-    def test_complex(self):
+    @parametrize_rgi_interp_methods
+    def test_complex(self, method):
         x, y, values = self._sample_2d_data()
         points = (x, y)
         values = values - 2j*values
 
         sample = np.array([[1, 2.3, 5.3, 0.5, 3.3, 1.2, 3],
                            [1, 3.3, 1.2, 4.0, 5.0, 1.0, 3]]).T
 
-        for method in ['linear', 'nearest']:
-            v1 = interpn(points, values, sample, method=method)
-            v2r = interpn(points, values.real, sample, method=method)
-            v2i = interpn(points, values.imag, sample, method=method)
-            v2 = v2r + 1j*v2i
-            assert_allclose(v1, v2)
+        v1 = interpn(points, values, sample, method=method)
+        v2r = interpn(points, values.real, sample, method=method)
+        v2i = interpn(points, values.imag, sample, method=method)
+        v2 = v2r + 1j*v2i
+        assert_allclose(v1, v2)
 
+    def test_complex_spline2fd(self):
         # Complex-valued data not supported by spline2fd
-        assert_warns(np.ComplexWarning, interpn, points, values,
-                     sample, method='splinef2d')
+        x, y, values = self._sample_2d_data()
+        points = (x, y)
+        values = values - 2j*values
 
-    def test_duck_typed_values(self):
+        sample = np.array([[1, 2.3, 5.3, 0.5, 3.3, 1.2, 3],
+                           [1, 3.3, 1.2, 4.0, 5.0, 1.0, 3]]).T
+        with assert_warns(np.ComplexWarning):
+            interpn(points, values, sample, method='splinef2d')
+
+    @pytest.mark.parametrize(
+        "method",
+        ["linear", "nearest"]
+    )
```

Comment:
Not sure if spline methods allow these "duck-typed values", and if we really need to worry about those. Can leave for a future enhancement if there is a user request.

Reply:
So I did look at adding the spline methods to this test but it required modifying the duck typed object to add things like ndims etc.

Reply:
Let's wait for a user request I'd say. This is a bit esoteric.